### PR TITLE
Enabling SQLite FTS5 extension (full text search)

### PR DESCRIPTION
**Pull request by geelen**
Builds upon #606.

Module is described here: https://www.sqlite.org/fts5.html

This adds a single build flag, `SQLITE_ENABLE_FTS5`, and 4 functions to the allowlist: `match`,`highlight`,`bm25` and `snippet`.

### CODE REVIEW

**Comment by elithrar** [comment]
Note: this is included in D1 alpha today and has users asking about it (we will document in dev docs).

The already-added JSON1, Math and (now) FTS5 should cover the major SQLite built-ins for now.

**Review by dom96 — APPROVED** [review]
Changes look good to me, but I don't have a lot of context on this. I'll approve but feel free to wait for Kenton if you think these changes need his approval.

**Comment by kentonv** [comment]
@dom96 Please leave any sqlite-related code reviews to me, it's important that I review these things as there are security implications here.

**Comment by kentonv** [comment]
I guess I'm OK with FTS5 but I have a bunch of comments over on #606 which this is based on.

**Pull request closed without merging**
